tests/func/test_add.py

```diff
@@ -59,6 +60,40 @@ def test_add(tmp_dir, dvc):
     }
 
 
+def test_add_output(tmp_dir, dvc):
+    (fd, fname) = tempfile.mkstemp()
+    os.write(fd, b"foo")
+    os.close(fd)
+
+    ret = main(["add", "--output", "foo", "bar", "baz"])
+    assert ret == 1
+
+    md5, _ = file_md5(fname)
+    stage = dvc.add(fname, output="foo")[0]
+
+    assert stage is not None
+
+    assert isinstance(stage, Stage)
+    assert os.path.isfile(stage.path)
+    assert len(stage.outs) == 1
+    assert len(stage.deps) == 0
+    assert stage.cmd is None
+    assert stage.outs[0].hash_info == HashInfo("md5", md5)
+    assert stage.md5 is None
+
+    assert load_yaml("foo.dvc") == {
+        "outs": [
+            {
+                "md5": "acbd18db4cc2f85cedef654fccc4a4d8",
+                "path": "foo",
+                "size": 3,
+            }
+        ],
+    }
+
+    os.unlink(fname)
```

Comment:
maybe just use tmp_path fixture or smth like that to avoid doing this manually and using file descriptors?